internal/build/phase_config_provider.go

```diff
@@ -122,14 +123,34 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 	}
 }
 
-func WithDaemonAccess() PhaseConfigProviderOperation {
+func WithDaemonAccess(dockerHost string) PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
-		bind := "/var/run/docker.sock:/var/run/docker.sock"
-		if provider.os == "windows" {
-			bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+		if dockerHost == "inherit" {
+			dockerHost = os.Getenv("DOCKER_HOST")
+		}
+		var bind string
+		if dockerHost == "" {
+			bind = "/var/run/docker.sock:/var/run/docker.sock"
+			if provider.os == "windows" {
+				bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+			}
+		} else {
+			switch {
```

Comment:
It also looks like `ssh` is supported (https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option), although I couldn't see how it's handled in the code: https://github.com/moby/moby/blob/41be7293f54f15dc04f024bf2b0f09e1a697208b/daemon/listeners/listeners_linux.go#L22-L56

I think that `ssh` is beyond the scope of this PR, though, if it's unclear how to handle it.

Reply:
Create an issues for this please.